src/main/kotlin/org/rust/ide/typing/paste/RsConvertJsonToStructCopyPasteProcessor.kt

```diff
@@ -0,0 +1,199 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.typing.paste
+
+import com.intellij.codeInsight.editorActions.CopyPastePostProcessor
+import com.intellij.codeInsight.editorActions.TextBlockTransferableData
+import com.intellij.json.JsonFileType
+import com.intellij.json.psi.*
+import com.intellij.openapi.application.runWriteAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.editor.RangeMarker
+import com.intellij.openapi.project.DumbService
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.Messages
+import com.intellij.openapi.ui.showYesNoDialog
+import com.intellij.openapi.util.Ref
+import com.intellij.openapiext.isUnitTestMode
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiErrorElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiFileFactory
+import org.rust.RsBundle
+import org.rust.ide.utils.template.buildAndRunTemplate
+import org.rust.lang.core.psi.RsFile
+import org.rust.lang.core.psi.RsPatWild
+import org.rust.lang.core.psi.RsPsiFactory
+import org.rust.lang.core.psi.RsStructItem
+import org.rust.lang.core.psi.ext.descendantOfTypeStrict
+import org.rust.lang.core.psi.ext.descendantsOfType
+import org.rust.openapiext.createSmartPointer
+import java.awt.datatransfer.DataFlavor
+import java.awt.datatransfer.Transferable
+
+class RsConvertJsonToStructCopyPasteProcessor : CopyPastePostProcessor<TextBlockTransferableData>() {
+    override fun collectTransferableData(
+        file: PsiFile?,
+        editor: Editor?,
+        startOffsets: IntArray?,
+        endOffsets: IntArray?
+    ): List<TextBlockTransferableData> = emptyList()
+
+    override fun extractTransferableData(content: Transferable): List<TextBlockTransferableData> {
+        try {
+            if (content.isDataFlavorSupported(DataFlavor.stringFlavor)) {
+                val text = content.getTransferData(DataFlavor.stringFlavor) as String
+                return listOf(PotentialJsonTransferableData(text))
+            }
+        } catch (e: Throwable) {
+        }
+        return emptyList()
+    }
+
+    override fun processTransferableData(
+        project: Project,
+        editor: Editor,
+        bounds: RangeMarker,
+        caretOffset: Int,
+        indented: Ref<in Boolean>,
+        values: List<TextBlockTransferableData>
+    ) {
+        if (DumbService.getInstance(project).isDumb) return
```

Comment:
Why do we need to check dumb mode here? Looks like we don't use indices anywhere

Reply:
It's probably not needed right now, but for more complex functionality (like `serde` lookup), it might be required, so I just inserted it proactively.